packages/trader/src/Modules/Trading/Components/Form/TradeParams/Turbos/__tests__/barrier-selector.spec.js

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import userEvent from '@testing-library/user-event';
+import BarrierSelector from '../barrier-selector';
+
+const mock_props = {
+    barrier_1: '16',
+    onChange: jest.fn(),
+    setHoveredBarrier: jest.fn(),
+    turbos_barrier_choices: ['16', '33', '40'],
+};
+
+jest.mock('Stores/connect.js', () => ({
+    __esModule: true,
+    default: 'mockedDefaultExport',
+    connect: () => Component => props => Component({ ...props, ...mock_props }),
+}));
+
+jest.mock('@deriv/shared', () => ({
+    ...jest.requireActual('@deriv/shared'),
+}));
+
+jest.mock('@deriv/components', () => {
+    const original_module = jest.requireActual('@deriv/components');
+    return {
+        ...original_module,
+        Icon: jest.fn(() => <div>IcCross</div>),
+    };
+});
+
+describe('Barriers Selector', () => {
+    let current_barrier;
+    beforeEach(() => {
+        render(<BarrierSelector {...mock_props} />);
+        current_barrier = screen.getByTestId('current_barrier');
+    });
+
+    it('The value of barrier_1 is chosen by default after first render', () => {
+        expect(screen.getByText(mock_props.barrier_1)).toBeInTheDocument();
+    });
+
+    it('The barrier list is hidden/collapsed by default', () => {
+        expect(screen.queryByText('Distance to spot')).not.toBeInTheDocument();
+    });
+
+    it('The barrier list is shown after cliking on current barrier', () => {
+        userEvent.click(current_barrier);
+
+        expect(screen.getByText('Distance to spot')).toBeInTheDocument();
+    });
+
+    it('All the available barriers from turbos_barrier_choices should be rendered in barrier list (when it can be shown)', () => {
+        userEvent.click(current_barrier);
+
+        mock_props.turbos_barrier_choices.forEach(barrier => expect(screen.getByTestId(barrier)).toBeInTheDocument());
```

Comment:
We can't replace `screen.getByTestId(barrier))` by `screen.getByText(barrier))` because at that moment we have a default chosen value (16) and the list of available barrier's options (16, 33, 40) rendered. If we use `getByText` the tests will crushed as there are two similar values (16). If we use `getAllByText` the tests will also crushed. 
We may use `parentNode` instead of `screen`, but it's a bag practice.